test/ArcxAnalyticsProvider.test.tsx

```diff
@@ -0,0 +1,150 @@
+import { cleanup, render, screen } from '@testing-library/react'
+import {
+  ArcxAnalyticsProvider,
+  ArcxAnalyticsProviderProps,
+  ArcxAnalyticxContext,
+  useArcxAnalytics,
+} from '../src'
+import { expect } from 'chai'
+import sinon from 'sinon'
+import * as postRequestModule from '../src/helpers/postRequest'
+import { ATTRIBUTION_EVENT, DEFAULT_SDK_CONFIG } from '../src/constants'
+import React from 'react'
+
+const IDENTITY_ID = 'test-dentity-id'
+const TEST_API_KEY = 'test-api-key'
+const CONFIG = {}
+
+const TestProvider = ({
+  children,
+  providerOverrides,
+}: {
+  children?: React.ReactNode
+  providerOverrides?: ArcxAnalyticsProviderProps
+}) => (
+  <ArcxAnalyticsProvider apiKey={TEST_API_KEY} config={CONFIG} {...providerOverrides}>
+    <ArcxAnalyticxContext.Consumer>
```

Comment:
Why do we need consumer here?

Reply:
to be able to access the `sdk` variable on the line 27

Reply:
i got it. Why don't add another level of `div` in `ChildTest`. Anyway it's looks good. I just didn't got why we put it here.

Reply:
It's because sometimes we want to test the provider alone and sometimes we want to simulate a component that uses the `useArcxaAnalytics()` hook